### PR TITLE
Fixes emissives, starlight, and a filterrific bug

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request
[Fixes emissive lighting, we never actually used the thing's layer, should have just been using float layers anyhow](https://github.com/tgstation/tgstation/commit/2af035d48742eb4337dbfabed699a9acef67d883)

[Fixes starlight not disabling fullbright](https://github.com/tgstation/tgstation/commit/7a4e3f0c9abacaf944a4be801d2d99daa72e9c50)

It used to, but tivi messed it up with area lighting. makes maps like kilo look WAY worse then they should (I stole this slightly from kapu)

[Fixes filterrific erroring when editing something with a color matrix](https://github.com/tgstation/tgstation/commit/72dfd4fd8fe4755858568ab7d80e31e146f49ea5)

It's not supported right now, mostly becuase it would require reworking
how filterrifc works out what window type to use, but at least this way
I can mess with the emissive plane without needing to recompile or do it
manually

## Why It's Good For The Game

Makes starlight actually pretty instead of just fullbright + the windows glow a bit, emissives working is good actually, and filterrific not being annoying makes me happy.

## Changelog
:cl:
fix: Starlight will actually show now, instead of just being replaced by fullbright most of the time
fix: Emissive lighting will once again work. Sorry lads
/:cl:

### CODE REVIEW

File: code/__HELPERS/filters.dm [hunk 1]
```diff
@@ -22,15 +22,13 @@ GLOBAL_LIST_INIT(master_filter_info, list(
 			"size" = 1
 		)
 	),
-	/* Not supported because making a proper matrix editor on the frontend would be a huge dick pain.
-		Uncomment if you ever implement it
+	// Not implenmented, but if this isn't uncommented some windows will just error
```
Comment (Fikou) [inline code/__HELPERS/filters.dm:25]: `implemented`
but also, this comment lacks context without the old one

File: code/__HELPERS/lighting.dm [hunk 2]
```diff
@@ -14,7 +14,7 @@
 	var/mutable_appearance/blocker = new()
 	blocker.icon = make_blocker.icon
 	blocker.icon_state = make_blocker.icon_state
-	blocker.layer = make_blocker.layer
+	blocker.layer = FLOAT_LAYER
```
Comment (Fikou) [inline code/__HELPERS/lighting.dm:17]:
```suggestion
```

Comment (Fikou) [reply code/__HELPERS/lighting.dm:17]: float is default value

File: code/__HELPERS/filters.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/__HELPERS/filters.dm:25]: Not implemented. Will detail though
```suggestion
	// Not implenmented, but if this isn't uncommented some windows will just error
	// Needs either a proper matrix editor, or just a hook to our existing one 
	// Issue is filterrific assumes variables will have the same value type if they share the same name, which this violates
	// Gotta refactor this sometime
```

Comment (Fikou) [reply code/__HELPERS/filters.dm:25]: i meant as in you made a typo

Comment (LemonInTheDark) [reply code/__HELPERS/filters.dm:25]:
```suggestion
	// Not implemented, but if this isn't uncommented some windows will just error
```

**Comment by Profakos** [comment]
Does the starlight change mean aurora borealis is no longer localizer entirely in the kitchen? Nearstation areas never for coloured while i was testing...

**Comment by LemonInTheDark** [comment]
> Does the starlight change mean aurora borealis is no longer localizer entirely in the kitchen? Nearstation areas never for coloured while i was testing...

yeet. the actual color of space won't change but the light it produces will

**Comment by Profakos** [comment]
Magnificent!

**Pull request merged**